pkg/kubelet/cm/deviceplugin/device_plugin_handler.go

```diff
@@ -113,22 +115,27 @@ func NewHandlerImpl(updateCapacityFunc func(v1.ResourceList)) (*HandlerImpl, err
 
 	mgr, err := NewManagerImpl(pluginapi.KubeletSocket, deviceManagerMonitorCallback)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to initialize device plugin manager: %+v", err)
+		return nil, fmt.Errorf("failed to initialize device plugin manager: %+v", err)
 	}
 
 	handler.devicePluginManager = mgr
 	handler.devicePluginManagerMonitorCallback = deviceManagerMonitorCallback
-
 	return handler, nil
 }
 
 // Start initializes podDevices and allocatedDevices information from checkpoint-ed state
 // and starts device plugin registration service.
 func (h *HandlerImpl) Start(activePods ActivePodsFunc) error {
 	h.activePods = activePods
+	dir, _ := filepath.Split(pluginapi.KubeletSocket)
+	var err error
+	h.store, err = utilstore.NewFileStore(dir, utilfs.DefaultFs{})
```

Comment:
Assuming that kubelet on node reboot will re-admit all pods, we can probably re-assign devices on reboot. Should we store checkpoints on tmpfs instead of durable storage and take the additional perf gains?

Reply:
Looking at the implementation of utilstore a bit more, I think whether we are writing to tmpfs or not actually depends on the directory path we pass in, so that may be orthogonal to this change.

Reply:
ok. Can you file an issue based on my previous comment @jiayingz ?

Reply:
Sure. Filed https://github.com/kubernetes/kubernetes/issues/56159